### ISSUE TITLE
npm package for node.js

### ISSUE THREAD

**Issue by nuest** [LibraryOfCongress/bagger-js#29]
Hi! Have you considered or tested your code for server-side JavaScript, i.e. an npm package for node.js?

**Comment by acdha** [comment]
I haven't – perhaps @edsu or @dbrunton might be familiar with any existing node.js implementations. I would want to reconsider some parts of the design because there would be far more straight-forward to do things like the crypto operations without having to work within the restricted browser environment, not to mention having things like high-performance hashing available directly without having to use asm.js in a web worker.

**Comment by edsu** [comment]
Yeah, I'm not sure. I had thought it might be useful to provide an atom/electron package for bagger-js to run it as a desktop app. But I doubt that's the usecase you are thinking of?

**Comment by acdha** [comment]
@edsu: @eikeon and I talked about using Electron very casually but I think that would be a great way to deprecate the old Java desktop app and avoid needing to maintain two separate codebases for both light and heavy users.

**Comment by edsu** [comment]
Yes, I was thinking the same thing :-D

**Comment by nuest** [comment]
Yes, @edsu that was what I was thinking of. However, the package bagit-js you link to is basically empty. I would like to develop a node application that handles bagit containers.

**Comment by nuest** [comment]
My apologies for leaving this thread kind of open ended. So I'm involved in a research project that has just started, where we will look into using BagIt to package both source code and runtimes (via  containers) of research analyses. Right now we're just starting, but I can imagine that we will implement our components with node.js, and if we do I'd really like to collaborate with you guys on this.

Until then: Are you aware of Google Summer of Code? Maybe you'd consider mentoring a student that could work on this topic. For information see https://summerofcode.withgoogle.com/

**Comment by acdha** [comment]
It looks like there are now multiple projects on NPM. Given how much this depends on browser APIs which aren’t relevant to server apps does it seem like there is much reason to keep this issue open?
